### PR TITLE
fix(docker) Add platform to docker-compose command

### LINKED ISSUE #5682: Docker pulls images built with wrong platform when using Rosetta emulator with m1 chip with arm64 arch
**Describe the bug**
Docker pulls images built with wrong cpu architecture when using rosetta emulator 2 as described [here](https://docs.docker.com/desktop/mac/apple-silicon/) on macos with m1 chip with arm64 arch.

**To Reproduce**
Steps to reproduce the behaviour:
Using mac pro with m1 chip
1. Launch a terminal that uses Rosetta emulator (in my case it is iterm)
3. Follow datahub quickstart guide and run `datahub docker quickstart`
4. Pulling both `neo4j` and `mysql` images doesn't work and shows errors
`ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries`
`ERROR: for neo4j  no matching manifest for linux/arm64/v8 in the manifest list entries`
`ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries`

**Expected behavior**
Expecting it to pull images but with amd64 arch and not `arm64.`

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: macos
 - DataHub CLI version: 0.8.43.2

**Root cause**
The problem here is that when using an `amd64` emulator the shell result of `uname -m `  is `x86_64` but 

> docker automatically selects the image variant that matches your OS and architecture

as [documented here ](https://docs.docker.com/desktop/mac/apple-silicon/) which will be `arm64` because [Docker desktop for M1 ](https://docs.docker.com/desktop/mac/apple-silicon/) chip is installed.

We end up pulling images with `arm64` arch but without overwriting `mysql` and `neo4j` docker compose services with `docker-compose.m1.yml` file as desired in `docker/dev.sh` script

**Potential solution**
We can pass `DOCKER_DEFAULT_PLATFORM` env variable with `docker-compose` and set it to the desired arch with `uname -m` command as proposed in this fix PR #5683

### PR DESCRIPTION
**Pull request by firasomrane**
Fixes #5682 
(+) changing docker -> docker_cli file name to follow cli files convention.

Tested executing `python docker_cli.py` (after adding `if __name__ == "__main__": quickstart()` at the end)
image

Also fixed `docker/dev.sh` file executing `./docker/dev.sh` 

image

With the fix it pulls `mysql` and `neo4j` images successfully with the right architecture.

Inspecting the image to verify the arch and looks good `docker image inspect mysql_image_id`

```
"Architecture": "amd64",
"Os": "linux",
"Size": 430582471,
"VirtualSize": 430582471,
...
```

### CODE REVIEW

**Review by pedro93 — COMMENTED** [review]
LGTM overall, let's wait for CI to be green before running.
@firasomrane did you test this locally on a machine?
If yes what OS and if not could you please follow https://datahubproject.io/docs/developers#deploying-local-versions to deploy your local changes and deploy datahub.

**Review by szalai1 — COMMENTED** [review]
@firasomrane thanks for contributing this back. Could you add the same logic to `docker/quickstart.sh` as well?

**Comment by szalai1** [comment]
tested this locally on my M2 mac. Works fine.

**Comment by firasomrane** [comment]
@pedro93  Thanks for the review.
@szalai1 The `docker/quickstart.sh` file is fixed now, with other files using the `docker-compose`

**Review by szalai1 — APPROVED** [review]

File: metadata-ingestion/src/datahub/entrypoints.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from datahub.cli.check_cli import check
 from datahub.cli.cli_utils import DATAHUB_CONFIG_PATH, write_datahub_config
 from datahub.cli.delete_cli import delete
-from datahub.cli.docker import docker
```
Comment (anshbansal) [inline metadata-ingestion/src/datahub/entrypoints.py:14]: Is this using a newer version of the python dependency. If yes, can you please update the `setup.py` with the minimum requirement version you tested it with?

**Review by RyanHolstien — APPROVED** [review]

File: metadata-ingestion/src/datahub/entrypoints.py (hunk 1 above)
Comment (firasomrane) [reply metadata-ingestion/src/datahub/entrypoints.py:14]: Could you please explain which python dependency you are asking about ?

Comment (pedro93) [reply metadata-ingestion/src/datahub/entrypoints.py:14]: The version of the docker Python lib you used when testing this change.

**Comment by anshbansal** [comment]
There are merge conflicts that need to be fixed

**Pull request merged**